scripts/validate-github-actions-pinned.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+set -e
+
+sha1_regex='^[a-f0-9]{40}$'
+sha256_regex='^[A-Fa-f0-9]{64}$'
+
+# Default values
+workflows_path=".github/workflows"
+dry_run=true
+debug=false
+action_has_error=false
+
+# Parse command-line arguments
```

Comment:
I think shift might not be doing anything in this case statement. You may need to use something like this for switch to work:

```
while [ "$#" -gt 0 ]; do
  case $1 in
```

Reply:
Right, I didn't see that. Fixed